src/enum_maps/side_controller.hpp

```diff
@@ -0,0 +1,49 @@
+/*
+	Copyright (C) 2008 - 2021
+	Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+	This program is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; either version 2 of the License, or
+	(at your option) any later version.
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY.
+
+	See the COPYING file for more details.
+*/
+
+#pragma once
+
+#include <array>
+#include <string>
+
+class side_controller
+{
+public:
+	enum TYPE { INVALID = 0, NONE = 1, HUMAN = 2, AI = 3, RESERVED = 4 };
+	static constexpr const char* const invalid = "invalid";
+	static constexpr const char* const none = "null";
+	static constexpr const char* const human = "human";
+	static constexpr const char* const ai = "ai";
+	static constexpr const char* const reserved = "reserved";
```

Comment:
Having an `enum` rather than an `enum class`, combined with reusing the same names (except for a different case) for `char*` consts sounds likely to cause bugs where the wrong case is used, but the compiler silently converts `char *` to int.

Reply:
Fixed.